sbor/src/schema/schema_validation/type_metadata_validation.rs

```diff
@@ -0,0 +1,185 @@
+                    return Err(SchemaValidationError::TypeMetadataHasMismatchingEnumDiscriminator)
+                };
+
+                let TypeMetadata {
+                    type_name,
+                    children,
+                } = variant_metadata;
+                validate_enum_variant_name(type_name.as_ref())?;
+                validate_fields_child_names(children, child_types.len())?;
+            }
+        }
+    }
+    Ok(())
+}
+
+fn validate_type_name(name: &str) -> Result<(), SchemaValidationError> {
+    if name.len() == 0 {
+        return Err(SchemaValidationError::InvalidTypeName {
+            message: "Type name cannot be empty".into(),
+        });
+    }
+    if name.len() > 100 {
+        return Err(SchemaValidationError::InvalidTypeName {
+            message: "Type name cannot be more than 100 characters".into(),
+        });
+    }
+    for char in name.chars() {
+        if !matches!(char, '0'..='9' | 'A'..='Z' | 'a'..='z' | '_') {
+            return Err(SchemaValidationError::InvalidTypeName {
+                // We need to validate this because we need to generate types from these names
+                // Rust is much less prescriptive for identifier names (see https://doc.rust-lang.org/reference/identifiers.html and https://unicode.org/reports/tr31/)
+                // But we can always loosen this later
+                message: "At present, types names must match [0-9A-Za-z_]+".into(),
+            });
+        }
+    }
+    Ok(())
+}
+
+fn validate_enum_variant_name(name: &str) -> Result<(), SchemaValidationError> {
+    if name.len() == 0 {
+        return Err(SchemaValidationError::InvalidEnumVariantName {
+            message: "Enum variant name cannot be empty".into(),
+        });
+    }
+    if name.len() > 100 {
+        return Err(SchemaValidationError::InvalidEnumVariantName {
+            message: "Enum variant name cannot be more than 100 characters".into(),
+        });
+    }
+    for char in name.chars() {
+        if !matches!(char, '0'..='9' | 'A'..='Z' | 'a'..='z' | '_') {
+            return Err(SchemaValidationError::InvalidEnumVariantName {
+                message: "At present, enum variant names must match [0-9A-Za-z_]+".into(),
+            });
+        }
+    }
+    Ok(())
+}
+
+fn validate_field_name(name: &str) -> Result<(), SchemaValidationError> {
+    if name.len() == 0 {
+        return Err(SchemaValidationError::InvalidFieldName {
+            message: "Field name cannot be empty".into(),
+        });
+    }
+    if name.len() > 100 {
+        return Err(SchemaValidationError::InvalidFieldName {
+            message: "Field name cannot be more than 100 characters".into(),
+        });
+    }
+    for char in name.chars() {
+        if !matches!(char, '0'..='9' | 'A'..='Z' | 'a'..='z' | '_') {
+            return Err(SchemaValidationError::InvalidFieldName {
+                message: "At present, field names must match [0-9A-Za-z_]+".into(),
+            });
+        }
+    }
```

Comment:
Possible to have a common function like `validate_ident()`?

Reply:
Sure - I guess they are the same here, although conceptually they could be different. Will commonise them 👍

Reply:
100% agree. Just for now all the different idents share the same format. The comment is more from a minimum code perspective.